src/Centreon/Domain/Monitoring/Resource.php

```diff
@@ -0,0 +1,469 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Monitoring;
+
+use Centreon\Domain\Monitoring\Icon;
+use Centreon\Domain\Monitoring\Resource;
+use Centreon\Domain\Monitoring\ResourceStatus;
+use Centreon\Domain\Monitoring\ResourceSeverity;
+use DateTime;
+
+/**
+ * Class representing a record of a resource in the repository.
+ *
+ * @package Centreon\Domain\Monitoring
+ */
+class Resource
+{
+    // Groups for serilizing
+    public const SERIALIZER_GROUP_MAIN = 'resource_main';
+    public const SERIALIZER_GROUP_PARENT = 'resource_parent';
+
+    // Types
+    public const TYPE_SERVICE = 'service';
+    public const TYPE_HOST = 'host';
+
+    /**
+     * @var string|null
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     */
+    private $type;
+
+    /**
+     * @var string|null
+     */
+    private $name;
+
+    /**
+     * @var string|null
+     */
+    private $detailsUrl;
+
+    /**
+     * @var \Centreon\Domain\Monitoring\Icon|null
+     */
+    private $icon;
+
+    /**
+     * @var \Centreon\Domain\Monitoring\Resource|null
+     */
+    private $parent;
+
+    /**
+     * @var \Centreon\Domain\Monitoring\ResourceStatus|null
+     */
+    private $status;
+
+    /**
+     * @var bool|null
```

Comment:
A Boolean can't be null. Just choose true or false.
You must specify a default value.

Reply:
It's because of the database and an inheritance from the state. So let it go.

Reply:
there is no inheritance mechanism for this field, so it should be true or false